### PR TITLE
Workflow runs show view displays incorrect SCM name in webhook tab 

### PR DESCRIPTION
**Pull request by rubhanazeem**
Gitea headers also contain headers of GitHub. To identify GitHub it is important to check the absence of gitea headers.

### CODE REVIEW

**Comment by rubhanazeem** [comment]
[Workflow runs show view displays incorrect SCM name in webhook tab (S)](https://trello.com/c/8WIRCQMd/2115-workflow-runs-show-view-displays-incorrect-scm-name-in-webhook-tab-s)

**Review by saraycp — APPROVED** [review]
[This part of the Gitea documentation](https://docs.gitea.io/en-us/webhooks/#event-information) confirms that.

**Comment by danidoni** [comment]
I believe we should trigger this use case in the specs and then confirm the fix against those specs.

File: src/api/app/models/workflow_run.rb [hunk 1]
```diff
@@ -110,7 +110,7 @@ def generic_event_type
   # together all the behaviour regarding GitHub and all the behaviour regarding
   # GitLab.
   def scm_vendor
-    if parsed_request_headers['HTTP_X_GITHUB_EVENT']
+    if parsed_request_headers['HTTP_X_GITHUB_EVENT'] && !parsed_request_headers['HTTP_X_GITEA_EVENT']
```
[the remark this answers is not in the document]
Comment (rubhanazeem) [reply src/api/app/models/workflow_run.rb]: Yeah, that will work. Having simpler conditions make it easy to understand. I've swapped the order

File: src/api/spec/models/workflow_run_spec.rb [hunk 2]
```diff
@@ -86,4 +88,24 @@
...
   end
+
+  describe '#scm_vendor' do
+    let(:workflow_run) { create(:workflow_run, request_headers: headers) }
+
+    context 'when event is from GitHub' do
+      let(:headers) { "HTTP_X_GITHUB_EVENT_TYPE: pull_request\nHTTP_X_GITHUB_EVENT: pull_request\nHTTP_ACCEPT: application/xml" }
+
+      it 'identifies GitHub' do
+        expect(workflow_run.scm_vendor).to be(:github)
```
Comment (danidoni) [inline src/api/spec/models/workflow_run_spec.rb:99]:
```suggestion
      subject { workflow_run.scm_vendor }
      it 'identifies GitHub' do
        expect(subject).to be(:github)
```
Then you can reuse the subject on the other context too. Using the subject block helps identifying what do you want to test.

**Review by danidoni — CHANGES_REQUESTED** [review]

**Review by danidoni — APPROVED** [review]

**Pull request merged**